optashift-employee-rostering-gwtui/src/main/java/org/optaplanner/openshift/employeerostering/gwtui/client/calendar/Calendar.java

```diff
@@ -0,0 +1,335 @@
+package org.optaplanner.openshift.employeerostering.gwtui.client.calendar;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+import elemental2.dom.CanvasRenderingContext2D;
+import elemental2.dom.HTMLCanvasElement;
+import elemental2.dom.MouseEvent;
+import com.google.gwt.user.client.Window;
+import com.google.gwt.user.client.ui.Panel;
+import org.jboss.errai.ioc.client.container.SyncBeanManager;
+import org.jboss.errai.ui.client.local.spi.TranslationService;
+import org.optaplanner.openshift.employeerostering.gwtui.client.common.ConstantFetchable;
+import org.optaplanner.openshift.employeerostering.gwtui.client.interfaces.DataProvider;
+import org.optaplanner.openshift.employeerostering.gwtui.client.interfaces.Fetchable;
+import org.optaplanner.openshift.employeerostering.gwtui.client.interfaces.HasTimeslot;
+
+public class Calendar<G extends HasTitle, I extends HasTimeslot<G>> {
```

Comment:
Probably could change it so that each panel is an `@Templated` Errai widget and remove all the UI code from the presenter. Calendar is a somewhat generic presenter, so the majority of the updating code is in other classes which is passed to it upon construction.